### PR TITLE
[Gluten-core] Delete spark_columnar_plugin_xxx on exit

### PR DESCRIPTION
**Pull request by Yohahaha**
## What changes were proposed in this pull request?

Use JVM delete on exit hook to delete tmp work dir.

## How was this patch tested?

manual test

(If this patch involves UI changes, please attach a screenshot; otherwise, remove this)

### CODE REVIEW

**Review by zhouyuan — APPROVED** [review]
👍

**Review by FelixYBW — APPROVED** [review]

**Comment by FelixYBW** [comment]
Good point

File: gluten-core/src/main/java/io/glutenproject/vectorized/JniWorkspace.java [hunk 1]
```diff
@@ -29,6 +30,13 @@ private JniWorkspace(String rootDir) {
       this.jniLibLoader = new JniLibLoader(workDir);
       this.jniResourceHelper = new JniResourceHelper(workDir);
       LOG.info("JNI workspace {} created in root directory {}", workDir, rootDir);
+      Runtime.getRuntime().addShutdownHook(new Thread(() -> {
+        try {
+          FileUtils.deleteDirectory(created.toFile());
+        } catch (Exception e) {
+          throw new RuntimeException(e);
+        }
+      }));
```
Comment (zhztheplayer) [inline gluten-core/src/main/java/io/glutenproject/vectorized/JniWorkspace.java:39]: nit: can we use FileUtils.deleteDirectoryOnExit(...) ?

Comment (Yohahaha) [reply gluten-core/src/main/java/io/glutenproject/vectorized/JniWorkspace.java:39]: `FileUtils.deleteDirectoryOnExit` is a private method.
I also try `FileUtils.forceDeleteOnExit`, but not work.

File: gluten-core/src/main/java/io/glutenproject/vectorized/JniWorkspace.java [hunk 2]
```diff
@@ -29,6 +30,13 @@ private JniWorkspace(String rootDir) {
       this.jniLibLoader = new JniLibLoader(workDir);
       this.jniResourceHelper = new JniResourceHelper(workDir);
       LOG.info("JNI workspace {} created in root directory {}", workDir, rootDir);
+      Runtime.getRuntime().addShutdownHook(new Thread(() -> {
```
Comment (jinchengchenghh) [inline gluten-core/src/main/java/io/glutenproject/vectorized/JniWorkspace.java:33]: Can you add it to `GlutenShutdownManager`

Comment (Yohahaha) [reply gluten-core/src/main/java/io/glutenproject/vectorized/JniWorkspace.java]: Nice solution.

**Review by jinchengchenghh — APPROVED** [review]

File: gluten-core/src/main/java/io/glutenproject/vectorized/JniWorkspace.java (hunk 1 above)
Comment (zhztheplayer) [reply gluten-core/src/main/java/io/glutenproject/vectorized/JniWorkspace.java]: Aha. Thanks for clarifying. But `FileUtils.forceDeleteOnExit`, this one should be suitable as long as it does what its doc says. But it's OK to keep the code as is.

**Comment by jinchengchenghh** [comment]
Can you help delete ` ls /tmp/gluten_9204097449682585560
libarrow.so.1100  libarrow.so.1100.0.0  libgluten.so  libparquet.so.1100  libparquet.so.1100.0.0  libvelox.so`?

**Pull request merged**